### PR TITLE
PIM-6967: Allow category panels to be resized

### PR DESCRIPTION
**Pull request by tamarasaurus**
**Description (for Contributor and Core Developer)**

This PR updates the category panels so that they can be resized. The resized width is saved in localStorage and restored when the panel is loaded. 

The panels that were updated: 

Asset picker (in the EE PR https://github.com/akeneo/pim-enterprise-dev/pull/3276)
asset-picker

Product category filter
product-categories

Categories index (and asset categories in EE)
categories-index

Categories edit (and asset categories in EE)
categories-edit

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | -
| Changelog updated                 | OK
| Review and 2 GTM                  | OK
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: CHANGELOG-2.0.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 - IM-825: allow concurrent AJAX requests by closing the session in a listener
 - PIM-6838: Display completeness panel after Attributes in the PEF
+- PIM-6967 - Allow category panels to be resized
```
Comment (pierallard) [inline CHANGELOG-2.0.md]: : instead of -

File: src/Pim/Bundle/EnrichBundle/Resources/views/CategoryTree/form.html.twig [hunk 2]
```diff
@@ -88,4 +89,15 @@
...
 <script type="text/javascript">
     window.flashMessages = JSON.parse('{{ app.session.flashbag.all|json_encode()|raw }}');
+
+    require(['pim/menu/resizable'], function(Resizable) {
+        $(function() {
+            Resizable.set({
+                minWidth: 300,
+                maxWidth: 500,
+                container: '.ui-resizable-container',
+                storageKey: 'category-tree-edit'
```
Comment (pierallard) [inline src/Pim/Bundle/EnrichBundle/Resources/views/CategoryTree/form.html.twig]: Please add this in the Asset Category Tree form.html.twig in EE plz

File: src/Pim/Bundle/EnrichBundle/Resources/views/CategoryTree/index.html.twig [hunk 3]
```diff
@@ -63,6 +65,13 @@
                         form.render();
                     }.bind(this));
                 });
+
+                Resizable.set({
```
Comment (pierallard) [inline src/Pim/Bundle/EnrichBundle/Resources/views/CategoryTree/index.html.twig]: Same remark :)

File: src/Pim/Bundle/UIBundle/Resources/public/less/components/ColumnConfigurator.less [hunk 4]
```diff
@@ -34,6 +34,18 @@
     &--gray {
       background: #F6F7FB; // TODO FIND COLOR
```
Comment (pierallard) [inline src/Pim/Bundle/UIBundle/Resources/public/less/components/ColumnConfigurator.less]: Lol,  what's this ?
Can you replace it by `@AknLightGray` plz ?
I think it's me :jetpack:

File: src/Pim/Bundle/UIBundle/Resources/public/less/components/ColumnConfigurator.less [hunk 5]
```diff
@@ -34,6 +34,18 @@
...
     }
+
+    &--resizable {
+      position: relative;
+
+      .ui-resizable-e {
```
Comment (pierallard) [inline src/Pim/Bundle/UIBundle/Resources/public/less/components/ColumnConfigurator.less]: nope, it's not BEM.
Please put this in jquery-ui shit (all with the &--resizable element)

File: src/Pim/Bundle/EnrichBundle/Resources/views/CategoryTree/form.html.twig (hunk 2 above)
Comment (tamarasaurus) [reply src/Pim/Bundle/EnrichBundle/Resources/views/CategoryTree/form.html.twig]: The resizer is already in the asset categories page in EE, there isn't a separate form.html.twig

File: src/Pim/Bundle/UIBundle/Resources/public/less/components/ColumnConfigurator.less (hunk 5 above)
Comment (tamarasaurus) [reply src/Pim/Bundle/UIBundle/Resources/public/less/components/ColumnConfigurator.less]: 💩

File: src/Pim/Bundle/EnrichBundle/Resources/views/CategoryTree/index.html.twig (hunk 3 above)
Comment (tamarasaurus) [reply src/Pim/Bundle/EnrichBundle/Resources/views/CategoryTree/index.html.twig]: already working

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/menu/resizable.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+         * with jQuery UI.
+         *
+         * @param {Object} options
+         */
+        set(options = {}) {
+            this.options = Object.assign(this.options, options);
+
+            const { maxWidth, minWidth, container } = this.options;
+
+            if (null === container) {
```
Comment (SamirBoulil) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/menu/resizable.js]: I don't get how this option variable here work ? in the destroy() you did this tough
` const container = $(this.options.container);`

Comment (SamirBoulil) [reply src/Pim/Bundle/EnrichBundle/Resources/public/js/menu/resizable.js]: aaah `+            const { maxWidth, minWidth, container } = this.options;` forget what I said :)

**Pull request merged**